### PR TITLE
(BKR-1013) Add Windows 2012R2 Japanese Template

### PR DESCRIPTION
**Pull request by glennsarti**
This commit adds the Japanese Windows 2012 R2 VMPooler template in preparation
of testing non-US English systems in Beaker.

### CODE REVIEW

**Comment by glennsarti** [comment]
Ran spec tests locally and passed.

File: lib/beaker-hostgenerator/data.rb [hunk 1]
```diff
@@ -859,6 +859,24 @@ def osinfo
             'template' => 'win-2012r2-x86_64'
           }
         },
+        'windows2012r2_ja-64' => {
```
Comment (Iristyle) [inline lib/beaker-hostgenerator/data.rb]: @waynr do you have any thoughts on the pattern proposed here for language specific VMs?  Glenn has run with `_ja` - but want to make sure that jives with any plans you had for the future.


Any input from you @joshcooper ?

Comment (nwolfe) [inline lib/beaker-hostgenerator/data.rb]: This seems fine, and makes sense to me.

**Comment by nwolfe** [comment]
> Ran spec tests locally and passed.

Man, how did you figure out how to work with these test fixtures?  It looks like you did the right thing 😄 

The workflow we have when adding a new platform is this:
1. Add the platform to data.rb
2. Run rake task to generate fixtures (like 500+ of them)
3. Git commit only the fixtures that have the new platform in them (and delete the other created/changed fixtures)

Nicely done

**Comment by nwolfe** [comment]
I'll let this sit for a bit to give Wayne, Josh, others a chance to look.

I can cut a release for this next week.

**Comment by glennsarti** [comment]
I mostly copied the Windows 2016 and OSX test fixtures.  Although I must say the the Windows test fixtures are a bit wonky.  We'll never provision a master on Windows and yet we test for that.

**Comment by glennsarti** [comment]
... Also ... There's a rake task to create the fixtures?!?!  Wish I knew that earlier!

**Comment by glennsarti** [comment]
As for language support, more is coming both for Windows and Linux.  It's more than just that, it's variation support e.g. different pre-built OS features/roles.

**Comment by nwolfe** [comment]
> Although I must say the the Windows test fixtures are a bit wonky. We'll never provision a master on Windows and yet we test for that.

That's a result of the fixture generation, which combines everything in every way.

**Comment by Iristyle** [comment]
@glennsarti did you run this copy of beaker-hostgenerator with a `TEST_TARGET=` and kick off a Puppet test run against Japanese Windows?

As long as that has checked out, I'm 👍 to merging.

**Comment by glennsarti** [comment]
@Iristyle I hand't good catch.

Have now and added results to the BKR ticket.  All good.

**Comment by Iristyle** [comment]
Ticket looks good, I'm merging.

**Pull request merged**